### PR TITLE
NIFI-6525 - Support JsonPath delete expressions

### PR DESCRIPTION
**Pull request by mans2singh**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

Support JsonPath delete expressions

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by pvillard31** [comment]
Hey @mans2singh - can you rebase your PR against master?

**Comment by mans2singh** [comment]
@pvillard31 - 

I've rebased to master and updated the code.  Please let me know if you have any additional advice.

Thanks

File: nifi-docs/src/main/asciidoc/expression-language-guide.adoc [hunk 1]
```diff
@@ -1576,6 +1576,58 @@ Expressions will provide the following results:
...
+  "children": [],
+  "spouse": null
+}
+..........
+
+.jsonPathDelete Examples
+|===================================================================
+| Expression | Value
+| `${myJson:jsonPathDelete('$.firstName')}` | `{"lastName":"Smith","age":25,"address":{"streetAddress":"21 2nd Street","city":"New York","state":"NY","postalCode":"10021-3100"},"phoneNumbers":[{"type":"home","number":"212 555-1234"},{"type":"office","number":"646 555-4567"}]}`
+| `${myJson:jsonPath('$.missing-path')}` | Returns original JSON document
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/expression-language-guide.adoc]:
```suggestion
| `${myJson:jsonPathDelete('$.missing-path')}` | Returns original JSON document
```

Comment (mans2singh) [reply nifi-docs/src/main/asciidoc/expression-language-guide.adoc]: Hi @pvillard31 - 

I've corrected the doc, please let me know if you have any other recommendations.  

Thanks for your comments.

**Comment by pvillard31** [comment]
Merged to master, thanks @mans2singh

**Comment by mans2singh** [comment]
Thanks @pvillard31 for your all your help.

**Pull request closed without merging**